h/jinja_extensions/navbar_data_admin.py

```diff
@@ -0,0 +1,109 @@
+from copy import deepcopy
+
+from h.security import Permission
+
+
+def navbar_data_admin(request):
+    """
+    Get the navigation bar displayed at the top of the admin page.
+
+    This is used in `templates/layouts/admin.html.jinja2`
+    """
+
+    for tab in deepcopy(_ADMIN_MENU):
+        if not request.has_permission(tab.pop("permission")):
+            continue
+
+        if route := tab.get("route"):
+            tab["url"] = request.route_url(route)
+
+        if children := tab.get("children"):
+            for child in children:
+                child["url"] = request.route_url(child["route"])
+
+        yield tab
+
+
+_ADMIN_MENU = [
+    {
+        "id": "index",
+        "permission": Permission.AdminPage.INDEX,
```

Comment:
It would be very nice to not have this duplicate the permissions listed in the views. I tried using [`view_execution_permitted()`](https://docs.pylonsproject.org/projects/pyramid/en/2.0-branch/api/security.html?highlight=has_permission#pyramid.security.view_execution_permitted), but I couldn't get it to work. It always says there's no matching view.

This comes with it's own problems as you'd have to construct a representative context object. So this would be swapping one issue for another.